### PR TITLE
Add support for extension definition in bolt.yml file

### PR DESCRIPTION
**Pull request by rossriley**
Allows locally autoloadable extensions to be added via the bolt.yml file.

### CODE REVIEW

**Comment by bobdenotter** [comment]
👍

File: app/bootstrap.php [hunk 1]
```diff
@@ -184,5 +186,27 @@
 
     }
 
+    $extensionArray = [];
+    foreach ((array)$config['extensions'] as $extensionClass) {
+        if (is_string($extensionClass)) {
+            $extensionClass = new $extensionClass();
+        }
+        if ($extensionClass instanceof ExtensionInterface) {
+            $extensionArray[] = $extensionClass;
```
Comment (GawainLynch) [inline app/bootstrap.php]: Can we invert and throw an exception as discussed in Slack?

**Comment by CarsonF** [comment]
Would it be better to do that logic inside of the extend closure? Wouldn't have to have a temp array. Maybe there would be some other benefit from doing it lazily?

**Comment by bobdenotter** [comment]
Sweet!

**Pull request merged**